translator/trace/zipkin/zipkinv2_to_traces.go

```diff
@@ -428,3 +431,16 @@ func extractInstrumentationLibrary(zspan *zipkinmodel.SpanModel) string {
 	}
 	return zspan.Tags[conventions.InstrumentationLibraryName]
 }
+
+func setTimestampsIfUnsetV2(span pdata.Span, destAttrs pdata.AttributeMap) {
+	// zipkin allows timestamp to be unset, but opentelemetry-collector expects it to have a value.
+	// unset gets converted to zero somewhere along the way.  If timestamp is unset,
+	// the conversion from this internal format back to zipkin format in zipkin exporter fails
+	if span.StartTime().AsTime().Unix() <= 0 {
```

Comment:
StartTime is already the Unix timestamp, so `span.StartTime().AsTime().Unix()` == `span.StartTime()`

Reply:
thank you.  This was my first attempt at any go.  I cleaned this up, and also fixed the implementation a little... this was actually incorrect.